src/main/java/io/vertx/gsoc2023/qotd/QoTDHandlers.java

```diff
@@ -23,36 +34,52 @@ public void getAllQuotes(RoutingContext ctx) {
         .end(quotes.encode()));
   }
 
+  public void postNewQuote(RoutingContext context) {
+    var payload = context.body().asJsonObject();
+    insertQuote(payload)
+      .onFailure(cause -> context.response().setStatusCode(400).end())
+      .onSuccess(quote -> {
+        eventBus.send(QUOTE_INSERTED, quote);
+        context.response().setStatusCode(201).end();
+      });
+  }
+
+  public void realtimeQuotes(RoutingContext context) {
+    context.request()
+      .toWebSocket()
+      .onSuccess(wsServer -> {
+        quoteInsertedConsumer.handler(message -> {
```

Comment:
The consumer should be unregistered whenever the websocket closes, otherwise they might accumulate and eventually consume all memory

Reply:
That is scary. I will be aware of that from now on.